OpenRA.Mods.Cnc/Traits/MadTank.cs

```diff
@@ -216,14 +148,94 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 
 				self.SetTargetLine(order.Target, Color.Red);
 				self.QueueActivity(new MoveAdjacentTo(self, order.Target, targetLineColor: Color.Red));
-				self.QueueActivity(new CallFunc(StartDetonationSequence));
+				self.QueueActivity(new DetonationSequence(self));
 			}
 			else if (order.OrderString == "Detonate")
 			{
 				if (!order.Queued)
 					self.CancelActivity();
 
-				self.QueueActivity(new CallFunc(StartDetonationSequence));
+				self.QueueActivity(new DetonationSequence(self));
+			}
+		}
+
+		class DetonationSequence : Activity
+		{
+			readonly Actor self;
+			readonly MadTank mad;
+			readonly WithFacingSpriteBody wfsb;
+			readonly ScreenShaker screenShaker;
+
+			int remainingTicks;
+
+			public DetonationSequence(Actor self)
+			{
+				this.self = self;
+				mad = self.Trait<MadTank>();
```

Comment:
Imho this should be passed as argument to the ctor.

Reply:
Done.